src/services/TechRecordsService.ts

```diff
@@ -331,6 +332,32 @@ class TechRecordsService {
     const notifiableAlterationIds = ["47", "48"];
     return notifiableAlterationIds.includes(testTypeId);
   }
+
+  public async updateEuVehicleCategory(vin: string, newEuVehicleCategory: EU_VEHICLE_CATEGORY): Promise<HTTPResponse | HTTPError> {
+    const techRecordWrapper: ITechRecordWrapper = await this.getTechRecordsList(vin, STATUS.ALL);
+    const nonArchivedTechRecord = techRecordWrapper.techRecord.filter((techRecord) => techRecord.statusCode !== STATUS.ARCHIVED);
+    if (nonArchivedTechRecord.length === 1) {
+      if(nonArchivedTechRecord[0].euVehicleCategory) {
+        return new HTTPResponse(200,HTTPRESPONSE.NO_EU_VEHCILE_CATEGORY_UPDATE_REQUIRED);
+      } else {
+        const statusCode = nonArchivedTechRecord[0].statusCode;
+        const newTechRecord: ITechRecord = {...nonArchivedTechRecord[0]};
+        nonArchivedTechRecord[0].statusCode = STATUS.ARCHIVED;
+        newTechRecord.euVehicleCategory = newEuVehicleCategory;
+        newTechRecord.statusCode = statusCode;
+        techRecordWrapper.techRecord.push(newTechRecord);
+        let updatedTechRecord;
+        try {
+          updatedTechRecord =  await this.techRecordsDAO.updateSingle(techRecordWrapper);
+        } catch (error) {
+          throw new HTTPError(500, HTTPRESPONSE.INTERNAL_SERVER_ERROR);
+        }
+        return new HTTPResponse(200, this.formatTechRecordItemForResponse(updatedTechRecord.Attributes as ITechRecordWrapper));
+      }
+    } else {
+      throw new HTTPError(400, "The vehicle has more than one non archived Tech record!");
```

Comment:
Could reverse the top level If statement, and throw this if length <> 1, in order to reduce nesting/complexity.